### PR TITLE
Pre-release updates

### PR DESCRIPTION
**Pull request by trallard**
## Description

This PR makes some updates needed to make a release

This pull request:

- Adds the `private` key to `package.json` needed after the addition of workspaces in #34 
- Bumps `conda-store-ui` version
- Bumps this package version
- Updates RELEASE.md to match the release checklist

## Pull request checklist

- [x] Did you test this change locally?
- [x] Did you update the documentation (if required)?
- [x] Did you add/update relevant tests for this change (if required)?

### CODE REVIEW

**Review by pavithraes — APPROVED** [review]
Thank you!

**Review by krassowski — CHANGES_REQUESTED** [review]
The CI failure is due to lock file being out of sync with dependencies, not due to lint failure. The lock file will need fixing.

File: RELEASE.md [hunk 1]
```diff
@@ -18,21 +18,27 @@ All the Python packaging instructions in the `pyproject.toml` file to wrap your
 > [!IMPORTANT]
 > There is no need to update the version in `pyproject.toml` as it is automatically updated by `hatch` when generating the package.
 
-3. To create a Python source package (`.tar.gz`) and the binary package (`.whl`) in the `dist/` directory, do:
+3. Optional - if there is a newer release of `conda-store-ui`, update the `conda-store-ui` dependency in the package.json file.
+
+   ```bash
+   yarn upgrade @conda-store/conda-store-ui@<version>
+   ````
```
Comment (krassowski) [inline RELEASE.md:25]: Thanks for a quick merge over on https://github.com/conda-incubator/jupyterlab-conda-store/pull/44. Now the thing needed to please the linter is fixing the extra backtick here:

```suggestion
   ```
```

Comment (krassowski) [reply RELEASE.md:25]: Note: my suggestion does not render properly because of some GitHub UI limitation. I do not suggest to deleted it.

Comment (trallard) [reply RELEASE.md]: was already fixing this lol

**Comment by krassowski** [comment]
The remaining issue is due to a bug in the latest `@types/react` which come from updated `conda-store-ui` version: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/66841. I opened https://github.com/conda-incubator/jupyterlab-conda-store/pull/46 with a fix.

**Review by krassowski — APPROVED** [review]
LGTM!

**Pull request merged**